### PR TITLE
fix(backend/kubernetes): Ensure valid naming of name field

### PR DESCRIPTION
**Pull request by NeilHanlon**
- Kubernetes v1.26 on VKE causes error when creating persistent volume claim because of uppercase characters in name field

This patch is trivial just in order to get it working - happy to implement differently.

The error in question:

```
The PersistentVolumeClaim "wp-01G1131R63FWBSPMA4ZAZTKLE-0-clone-0" is invalid: metadata.name: Invalid value: "wp-01G1131R63FWBSPMA4ZAZTKLE-0-clone-0": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
```

### CODE REVIEW

File: pipeline/backend/kubernetes/utils.go [hunk 1]
```diff
@@ -10,8 +12,16 @@ import (
...
 
+var dnsPattern = regexp.MustCompile(`[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*`)
+
 func dnsName(i string) string {
-	return strings.Replace(i, "_", "-", -1)
+	res := strings.Replace(strings.ToLower(i), "_", "-", -1)
+
+	match := dnsPattern.MatchString(res)
+	if !match {
+		log.Panic("name not valid kubernetes DNS name")
```
Comment (6543) [inline pipeline/backend/kubernetes/utils.go:22]: dont Panic make the dnsName func return an error to report up the chain ...

and create a new error type .. `var Name = errors.New(...)`

File: pipeline/backend/kubernetes/utils.go [hunk 2]
```diff
@@ -10,8 +12,16 @@ import (
 	"k8s.io/client-go/tools/clientcmd"
 )
 
+var dnsPattern = regexp.MustCompile(`[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*`)
```
Comment (6543) [inline pipeline/backend/kubernetes/utils.go:15]:
```suggestion
var (
  dnsPattern = regexp.MustCompile(`[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*`)
  ErrDnsPatternInvalid = errors.New("name not valid kubernetes DNS name")
)
```
:point_up: this + `make format`

File: pipeline/backend/kubernetes/utils.go (hunk 1 above)
Comment (6543) [inline pipeline/backend/kubernetes/utils.go:22]:
```suggestion
		return "", ErrDnsPatternInvalid
```

**Comment by 6543** [comment]
also adding a smal unit test might be nice :)

if you need help just tell

**Comment by NeilHanlon** [comment]
Thank you for the comments! I've introduced a new commit which addresses the review items.

**Comment by lafriks** [comment]
ULID that is generated it volume name can be lowercased as ULIDs are case insensitive

**Pull request merged**